oclint-scripts/build

```diff
@@ -29,8 +29,6 @@ def build_command(llvm_root, is_release, module_extras, source_path):
     cmd_build = cmake.builder(source_path)
     if is_release:
         cmd_build.release_build()
-    if environment.is_unix():
-        cmd_build.use_clang_compiler(llvm_root)
```

Comment:
Here is what I am suggesting:

Let's introduce a new argument to this script

```python
arg_parser.add_argument('-use-system-compiler', '--use-system-compiler', action="store_true")
```

And here, we say
```python
if environment.is_unix() and not args.use_system_compiler:
    cmd_build.use_clang_compiler(llvm_root)
```

Same idea applies to `test` and `dogFooding` scripts.